pennylane/ops/functions/equal.py

```diff
@@ -85,12 +86,12 @@ def equal(
     if op1.__class__ is not op2.__class__:
         return False
 
-    if op1.__class__ is MeasurementProcess:
-        return equal_measurements(op1, op2)
-
-    if op1.__class__ is ShadowMeasurementProcess:
+    if isinstance(op1, ShadowMeasurementProcess):
         return equal_shadow_measurements(op1, op2)
 
+    if isinstance(op1, MeasurementProcess):
+        return equal_measurements(op1, op2)
```

Comment:
Any reason the order was swapped here?

Reply:
Because `ShadowMeasurementProcess` is an instance of `MeasurementProcess`